overlord/snapstate/snapmgr.go

```diff
@@ -259,7 +271,15 @@ func (snapst *SnapState) CurrentInfo() (*snap.Info, error) {
 	if cur == nil {
 		return nil, ErrNoCurrent
 	}
-	return readInfo(cur.RealName, cur, 0)
+
+	name := snap.InstanceName(cur.RealName, snapst.InstanceKey)
+	info, err := readInfo(name, cur, 0)
+	if err != nil {
+		return nil, err
+	}
+	info.InstanceKey = snapst.InstanceKey
```

Comment:
I might be missing a bigger picture, but looking at `readInfo` we call `snap.ReadInfo` and it sets `InstanceKey` there. Are we deliberately overwriting this here?

Reply:
Thanks for spotting this. The pieces in `snap` are part of the same branch but were landed independently. Wil push a fix.